app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayMiddleware.kt

```diff
@@ -0,0 +1,72 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.tabstray
+
+import androidx.annotation.VisibleForTesting
+import mozilla.components.lib.state.Middleware
+import mozilla.components.lib.state.MiddlewareContext
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.components.metrics.MetricController
+
+/**
+ * [Middleware] that reacts to various [TabsTrayAction]s.
+ *
+ * @property metrics reference to the configured [MetricController] to record general page load events.
+ */
+class TabsTrayMiddleware(
+    private val metrics: MetricController
+) : Middleware<TabsTrayState, TabsTrayAction> {
+
+    private var shouldReportInactiveTabMetrics: Boolean = true
+    private var shouldReportSearchGroupMetrics: Boolean = true
+
+    override fun invoke(
+        context: MiddlewareContext<TabsTrayState, TabsTrayAction>,
+        next: (TabsTrayAction) -> Unit,
+        action: TabsTrayAction
+    ) {
+        next(action)
+
+        when (action) {
+            is TabsTrayAction.UpdateInactiveTabs -> {
+                if (shouldReportInactiveTabMetrics) {
+                    shouldReportInactiveTabMetrics = false
+                    metrics.track(Event.InactiveTabsCountUpdate(action.tabs.size))
+                    metrics.track(Event.TabsTrayHasInactiveTabs(action.tabs.size))
```

Comment:
It seems more obvious with these two calls side-by-side now that there is no need for both of them to exist or if we've misinterpreted our we're supposed to collect the probes.

Please file an issue to follow-up on this.

Reply:
For context, the first was made with an Extra, and the second was made under #22155 to make that probe a quantity metric with the intent of later removing the first. @Mugurell, is there an issue to track this yet?